principles.md

```diff
@@ -0,0 +1,64 @@
+# Principles for software development standards
+
+These are general principles that underpin the software development guidance and standards.
+
+## Background
+
+There are a number of factors that can influence the decision to adopt a particular development standard. The purpose of these principles is to condense all of those factors into a simple set of statements that can be used to help with those decisions.
+
+## Principles
+
+### Don't re-invent the wheel
+
+> We adopt standards, guidance and approaches that are already in use in the industry or elsewhere in government
+
+- This accords with the general [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)
+- It saves us effort
+
+### Do the minimum necessary
+
+> We only develop guidance and standards where there is a rationale for it
+
+- This accords with the general [principle of keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
+- We don't needlessly constrain work by creating unnecessary standards
+- It saves us effort
+
+### Start small
+
+> We create a minimum viable product and then iteratively improve it
+
+- We quickly see if something works
+- It is easier to respond when things change
+- It prevents us from [trying to predict the future](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)
+- It helps us do the minimum necessary
+
+### Be open
+
+> We work in the open, providing the maximum amount of access possible
+
+- [More eyes makes things better](https://www.gov.uk/guidance/government-design-principles#make-things-open-it-makes-things-better)
+- People can buy-in more easily
+- It helps us do the minimum necessary
+
+### Minimise work in progress
+
+> We don't try to do too much at once
+
+- It's better to have one thing that can be used than many that can't
+
+### Be reasonable
+
+> We understand that standards represent a snapshot of our best understanding at a point in time and we
+explicitly articulate this in all of our standards
+
+- This allows us to make sensible, value-based judgements about when it is appropriate to retro-fit standards
+- It ensures clarity by mandating that all standards must define the period of time that they are effective
```

Comment:
I think we just need to agree how we handle those situations when a standard changes. We're looking to adopt industry standards where possible (don't reinvent the wheel) and as such _they will change_ anyway.

I'd rather we approach standards on this basis. Rather than trying to cross-reference projects to versioned standards (which also means needing a discussion about versioning) I'd prefer we just accept an approach where most of our projects will be behind current standards to a greater or lesser extent.

So where active development is taking place as long as some form of plan is in place to move to the current standard that is sufficient, even if that's just agreeing to adopt something like the [boy scout rule](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html).

Reply:
Yes, I should have avoided use of the word "version" but I didn't have a better shorthand for "the standard as it was written at the time that it was adopted".

The point I was trying to make is that the standards need to be clear and open about when they became effective (especially, I suppose, when they change) and when they cease to be effective.

So we can say
- "we chose to adopt this industry standard on date X"
- "it changed in this way on date Y and we're cool about that"
- "it changed in a hideous, unrecognisable way on date Z so we adopted another standard instead, but that's OK because we will know that anything that was in train before date Z will have been following the previous standard and shouldn't be judged too harshly for not yet following the current one"